hathor/consensus.py

```diff
@@ -249,17 +249,40 @@ def update_voided_info(self, block: Block) -> None:
                 else:
                     storage._best_block_tips = [blk.hash for blk in heads]
                     # XXX Is it safe to select one of the heads?
-                    best_block = heads[0]
-                    assert best_block.hash is not None
-                    best_meta = best_block.get_metadata()
-                    self.log.debug('index previous best block', height=best_meta.height, block=best_block.hash_hex)
-                    storage.add_new_to_block_height_index(best_meta.height, best_block.hash)
+                    # best_block = heads[0]
+                    # assert best_block.hash is not None
+                    # best_meta = best_block.get_metadata()
+                    # self.log.debug('index previous best block', height=best_meta.height, block=best_block.hash_hex)
+                    # storage.add_new_to_block_height_index(best_meta.height, best_block.hash, block.timestamp)
 
         # Uncomment the following lines to check that the cache update is working properly.
         # You shouldn't run this test in production because it dampens performance.
         #     v = storage.get_best_block_tips(skip_cache=True)
         #     assert v == storage._best_block_tips
 
+    def update_block_height_cache_new_chain(self, height: int, block: Block) -> None:
```

Comment:
It seems it was already implemented in the transaction storage. Can you check whether they are the same?

```bash
❯ grep -RIi update_block_height_cache_new_chain hathor
hathor/transaction/storage/transaction_storage.py:    def update_block_height_cache_new_chain(self, height: int, block: Block) -> None:
hathor/consensus.py:                        storage.update_block_height_cache_new_chain(meta.height, block)
hathor/consensus.py:    def update_block_height_cache_new_chain(self, height: int, block: Block) -> None:
```

Reply:
You're right, this got duplicated after a rebase.

Reply:
You're right, this got duplicated after a rebase.

Reply:
Fixed in #292 .